### PR TITLE
Filter released ticket by date range

### PR DESCRIPTION
**Pull request by andistuder**
As a dashboard user, I want to filter tickets by deploy date.

### CODE REVIEW

**Comment by 0mega** [comment]
:+1:

**Comment by andistuder** [comment]
ping @LogaJ

**Comment by LogaJ** [comment]
Will give thumbs up once modifications have been made from @steveweet's feedback.

File: app/models/repositories/released_ticket_repository.rb [hunk 1]
```diff
@@ -42,6 +46,25 @@ def apply(event)
 
     attr_reader :feature_review_factory, :git_repository_loader
 
+    def filter_tickets_by_date(from_date, to_date)
+      query = build_query(from_date, to_date)
+      store.find_by_sql(query).map(&:key) if query
+    end
+
+    def build_query(from_date, to_date)
+      query_prefix =  'SELECT DISTINCT ON(key) key ' \
```
Comment (LogaJ) [inline app/models/repositories/released_ticket_repository.rb]: Neat.

File: app/models/repositories/released_ticket_repository.rb [hunk 2]
```diff
@@ -64,24 +87,27 @@ def snapshot_deploy_event(event)
...
-             GitRepository::CommitNotFound => e
+      rescue GitRepositoryLoader::NotFound, GitRepository::CommitNotValid, GitRepository::CommitNotFound => e
         log_warning(e, event)
       end
     end
 
     def update_ticket_deploy_data(event, commit)
       tickets_for_versions(commit.associated_ids).each do |ticket_record|
-        next if duplicate_deploy?(ticket_record.deploys, event)
+        update_record(ticket_record, event, commit)
```
Comment (andistuder) [inline app/models/repositories/released_ticket_repository.rb]: revert here pls.

**Comment by LogaJ** [comment]
LGTM.

**Pull request merged**